src/routes.py

```diff
@@ -292,28 +297,51 @@ async def post_operation(
         entrypoint_name = operation["parameters"]["entrypoint"]
 
         try:
-            crud.get_entrypoint(db, str(contract.address), entrypoint_name)
+            entrypoint = crud.get_entrypoint(db, str(contract.address), entrypoint_name)
+            if not entrypoint.is_enabled:
+                raise EntrypointDisabled()
         except EntrypointNotFound:
             logging.warning(f"Entrypoint {entrypoint_name} is not found")
             raise HTTPException(
                 status_code=status.HTTP_400_BAD_REQUEST,
                 detail=f"Entrypoint {entrypoint_name} is not found",
             )
+        except EntrypointDisabled:
+            logging.warning(f"Entrypoint {entrypoint_name} is disabled.")
+            raise HTTPException(
+                status_code=status.HTTP_403_FORBIDDEN,
+                detail=f"Entrypoint {entrypoint_name} is disabled.",
+            )
 
     try:
         # Simulate the operation alone without sending it
         # TODO: log the result
         op = tezos.simulate_transaction(call_data.operations)
+
         logging.debug(f"Result of operation simulation : {op}")
+
         op_estimated_fees = [(int(x["fee"]), x["destination"]) for x in op.contents]
         estimated_fees = tezos.group_fees(op_estimated_fees)
-        logging.debug(f"Estimated fees for {op.hash()}: {estimated_fees}")
+
+        logging.debug(f"Estimated fees: {estimated_fees}")
+
         if not tezos.check_credits(db, estimated_fees):
             logging.warning(f"Not enough funds to pay estimated fees.")
-            raise HTTPException(
-                status_code=status.HTTP_400_BAD_REQUEST, detail="Not enough funds."
+            raise NotEnoughFunds(
+                f"Estimated fees : {estimated_fees[str(contract.address)]} mutez"
             )
+        if not tezos.check_calls_per_month(db, contract.id):  # type: ignore
+            logging.warning(f"Not enough calls for this month.")
+            raise NotEnoughCallsForThisMonth()
+
```

Comment:
That's a bad error message, as it means nothing to the user and is, in fact, saying the opposite of what it means.